src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -37,12 +46,45 @@ public LogicManager(Model model, Storage storage) {
         gradPadParser = new GradPadParser();
     }
 
+    private void assignStalledComponents(Command command, String commandText) {
+        stalledCommand = command;
+        stalledCommandText = commandText.trim();
+    }
+
     @Override
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
+        if (stalledCommand instanceof Command) {
+            if (!commandText.equalsIgnoreCase("yes")) {
+                stalledCommand = null;
+                return new CommandResult(MESSAGE_CONFIRMATION_CANCEL
+                        + String.format("\"%s\"", stalledCommandText));
+            }
+        }
+
         CommandResult commandResult;
         Command command = gradPadParser.parseCommand(commandText);
+
+        if (command instanceof ClearCommand) {
+            assignStalledComponents(command, commandText);
+            return new CommandResult(ClearCommand.MESSAGE_CONFIRMATION + MESSAGE_CONFIRMATION_SYNTAX);
+        } else if (command instanceof DeleteCommand) {
+            assignStalledComponents(command, commandText);
+            Module moduleToBeDeleted = ((DeleteCommand) command).getModuleToDelete(model);
+            return new CommandResult(DeleteCommand.MESSAGE_CONFIRMATION + moduleToBeDeleted
+                    + MESSAGE_CONFIRMATION_SYNTAX);
+        }
```

Comment:
You mentioned that you were concerned about messiness earlier but I think this implementation is fine actually! 😀 

But if we want to, we can abstract this out into a `stallComponent` fxn that handles the stalling. It might take in a `Command`, does the necessary stalling/operations, then returns the `CommandResult`. This will make things more future-proof cos we don't know how future commands need confirmation, and we don't wanna keep stuffing everything into `execute`
```suggestion
        if (command instanceof ClearCommand || command instanceof DeleteCommand) {
                return stallCommand(command); // returns the CommandResult
        }
```